src/DBAL/QueryFunction.php

```diff
@@ -0,0 +1,357 @@
+    {
+        $args = array_values($arguments);
+        $params = $args[0];
+        if (!is_array($params)) {
+            $params = [$params];
+        }
+
+        // Map camelCase function names to SQL function names
+        $func_name = match (true) {
+            // if the name is in camelCase, convert camelCase to snake_case
+            (bool) preg_match('/[A-Z]/', $name) => preg_replace('/(?<!^)[A-Z]/', '_$0', $name),
+            // Place any special formatting cases here
+            default => $name,
+        };
+        $func_name = strtoupper($func_name);
+        return self::getExpression($func_name, $params, $args[1] ?? null);
+    }
+
+    /**
+     * Build an ADDDATE SQL function call
+     * @param string|QueryExpression $date Date to add interval to
+     * @param string|QueryExpression $interval Interval to add
+     * @param string $interval_unit Interval unit
+     * @param string|null $alias Function result alias (will be automatically quoted)
+     * @return QueryExpression
+     */
+    public static function addDate(string|QueryExpression $date, string|QueryExpression $interval, string $interval_unit, ?string $alias = null): QueryExpression
+    {
+        global $DB;
+        $date = $date instanceof QueryExpression ? $date : $DB::quoteName($date);
+        $interval = $interval instanceof QueryExpression ? $interval : $DB::quoteName($interval);
+        $exp = sprintf('DATE_ADD(%s, INTERVAL %s %s)', $date, $interval, strtoupper($interval_unit));
+        return new QueryExpression($exp, $alias);
+    }
+
+    /**
+     * Build an IF SQL function call
+     * @param string|QueryExpression|array $condition Condition to test
+     * @param string|QueryExpression $true_expression Expression to return if condition is true
+     * @param string|QueryExpression $false_expression Expression to return if condition is false
+     * @param string|null $alias Function result alias (will be automatically quoted)
+     * @return QueryExpression
+     */
+    public static function if(string|QueryExpression|array $condition, string|QueryExpression $true_expression, string|QueryExpression $false_expression, ?string $alias = null): QueryExpression
+    {
+        if (is_array($condition)) {
+            $condition = new QueryExpression((new DBmysqlIterator(null))->analyseCrit($condition));
+        }
+        return self::getExpression('IF', [$condition, $true_expression, $false_expression], $alias);
+    }
+
+    /**
+     * Build an IFNULL SQL function call
+     * @param string|QueryExpression $expression Expression to check
+     * @param string|QueryExpression $value Value to return if expression is null
+     * @param string|null $alias Function result alias (will be automatically quoted)
+     * @return QueryExpression
+     */
+    public static function ifNull(string|QueryExpression $expression, string|QueryExpression $value, ?string $alias = null): QueryExpression
```

Comment:
Maybe it should follow naming rule applied to magic methods.
Same remark for `nullIf`, `timestampDiff` and `dateDiff`.
```suggestion
    public static function ifnull(string|QueryExpression $expression, string|QueryExpression $value, ?string $alias = null): QueryExpression
```